iml-api/src/graphql.rs

```diff
@@ -438,6 +443,26 @@ impl QueryRoot {
     }
 }
 
+struct SnapshotIntervalName {
+    id: i32,
+    fs_name: String,
+    timestamp: DateTime<Utc>,
+}
+
+fn parse_snapshot_name(name: &str) -> Option<SnapshotIntervalName> {
+    match name.split("-").collect::<Vec<&str>>().as_slice() {
+        [id, fs, ts] => match ts.parse::<DateTime<Utc>>() {
+            Ok(ts) => Some(SnapshotIntervalName {
+                id: id.parse::<i32>().expect("get id from name"),
```

Comment:
Why panic here instead of returning `None`?

Reply:
If it isn't in the right format shouldn't we fail fast? If we return None for the id how should we handle it?

Reply:
I think we need to assume that the pieces are as expected or it's not been sent by us. 

In other words this should return `None` here instead of panicking.

Reply:
Ok, I understand. How does this look?
```
fn parse_snapshot_name(name: &str) -> Option<SnapshotIntervalName> {
    match name.split("-").collect::<Vec<&str>>().as_slice() {
        [id, fs, ts] => match ts.parse::<DateTime<Utc>>() {
            Ok(ts) => {
                id.parse::<i32>()
                    .map(|id| {
                        SnapshotIntervalName {
                            id,
                            fs_name: fs.to_string(),
                            timestamp: ts,
                        }
                    })
                    .ok()
            },
            Err(_) => None,
        },
        _ => None,
    }
}
```

Reply:
How about this:

```
fn parse_snapshot_name(name: &str) -> Option<SnapshotIntervalName> {
    match name.split("-").collect::<Vec<&str>>().as_slice() {
        [id, fs, ts] => {
            let ts = ts.parse::<DateTime<Utc>>().ok()?;
            let id = id.parse::<i32>().ok()?;

            Some(SnapshotIntervalName {
                id,
                fs_name: fs.to_string(),
                timestamp: ts,
            })
        }
        _ => None,
    }
}
```

Reply:
Sure 👍